### PR TITLE
Fix documentation problems.

### PR DESCRIPTION
**Pull request by coryan**
This is the first in a series of fixes for the documentation.

You can look at the effect of these fixes in:

https://coryan.github.io/google-cloud-cpp/latest/storage/

Though for the moment only this is changing:

https://coryan.github.io/google-cloud-cpp/latest/storage/classgoogle_1_1cloud_1_1storage_1_1v0_1_1Client.html

### CODE REVIEW

File: google/cloud/storage/client.h [hunk 1]
```diff
@@ -47,7 +47,9 @@ inline namespace STORAGE_CLIENT_NS {
  * copy-construction are also relatively low-cost operations, they should be
  * comparable to copying a few shared pointers. The first request (or any
  * request that requires a new connection) incurs the cost of creating the
- * connection and authenticating with the service.
+ * connection and authenticating with the service. Note that access tokens need
+ * to be refreshed before they expire, , the library
```
Comment (devjgm) [inline google/cloud/storage/client.h]: Seems to be an extra space and comma here.

File: google/cloud/storage/client.h [hunk 2]
```diff
@@ -128,6 +148,17 @@ class Client {
 
   /**
    * Creates the default client type given the credentials and policies.
+   *
+   * @param credentials a set of credentials to initialize the `ClientOptions`.
+   * @param policies the client policies, these control the behavior of the
+   *   client, for example, how to backoff when a operation needs to be retried,
```
Comment (devjgm) [inline google/cloud/storage/client.h]: nit: a -> an

File: google/cloud/storage/client.h (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/client.h]: Fixed.

File: google/cloud/storage/client.h (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/client.h]: Fixed (here and one other place).

File: google/cloud/storage/client.h [hunk 3]
```diff
@@ -2075,6 +2117,7 @@ class Client {
   //@}
 
   //@{
+  /// @name Signed URL support operations.
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/storage/client.h]: Fixed.

File: google/cloud/storage/client.h [hunk 4]
```diff
@@ -83,9 +85,27 @@ inline namespace STORAGE_CLIENT_NS {
...
+ * @code
+ * namespace gcs = google::cloud::storage;
+ * gcs::Client client = ...;
+ * google::cloud::StatusOr<gcs::BucketMetadata> bucket_metadata =
+ *     client.GetBucketMetadata("my-bucket");
+ * if (!bucket_metadata.ok()) {
+ *   std::cerr << "Error getting metadata for my-bucket: "
+ *             << bucket_metadata.status() << std::endl;
+ *   return;
+ * }
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/storage/client.h]: Done.

File: google/cloud/storage/client.h [hunk 5]
```diff
@@ -83,9 +85,27 @@ inline namespace STORAGE_CLIENT_NS {
...
- * using `StatusOr<T>` to handle errors.
+ * contains the expected result. Please consult the `StatusOr<T>`
+ * [documentation](https://github.com/googleapis/google-cloud-cpp/blob/master/google/cloud/status_or.h)
+ * for more details.
+ *
+ * @code
+ * namespace gcs = google::cloud::storage;
+ * gcs::Client client = ...;
+ * google::cloud::StatusOr<gcs::BucketMetadata> bucket_metadata =
+ *     client.GetBucketMetadata("my-bucket");
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/storage/client.h]: Done.

File: google/cloud/storage/client.h (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/client.h]: Sorry, I missed @frankyn comment.  Does this work?

**Comment by coryan** [comment]
PTAL.

**Pull request merged**